src/selectedtests/task_mappings/mappings.py

```diff
@@ -62,57 +64,66 @@ def create_task_mappings(
             before_date=before_date,
         )
         log.info("Starting to generate task mappings")
-        project_versions: Generator[Version] = evg_api.versions_by_project(evergreen_project)
+        project_versions = evg_api.versions_by_project_time_window(
```

Comment:
I'm switching this to use the `*_time_window` iterator, which lets us remove some of the logic form this code.